### PR TITLE
Sr update rights

### PR DESCRIPTION
**Pull request by simen-rekkedal**
## Related Issue(s)
- #{issue number}

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Authentication/Controllers/SystemRegisterController.cs [hunk 1]
```diff
@@ -116,4 +116,40 @@ public async Task<ActionResult> CreateRegisteredSystem([FromBody] RegisterSystem
...
     }
-}
+
+    /// <summary>
+    /// Updates the rights on a registered system
+    /// </summary>
+    /// <param name="rights">A list of rights</param>
+    /// <param name="systemId">The human readable string id</param>
+    /// <returns>true if changed</returns>
+    [HttpPatch("system/{systemId}/rights")]
```
Comment (Alxandr) [inline src/Authentication/Controllers/SystemRegisterController.cs:126]: httpput

File: src/Authentication/Controllers/SystemRegisterController.cs [hunk 2]
```diff
@@ -116,4 +116,40 @@ public async Task<ActionResult> CreateRegisteredSystem([FromBody] RegisterSystem
...
+    //[Authorize(Policy = AuthzConstants.POLICY_SCOPE_SYSTEMREGISTER_WRITE)]
+    public async Task<ActionResult> UpdateRightsOnRegisteredSystem([FromBody] List<Right> rights, string systemId)
+    {
+        bool success = await _systemRegisterService.UpdateRightsForRegisteredSystem(rights, systemId);
+        if (!success)
+        {
+            return BadRequest();
+        }
+
+        return Ok(true);
```
Comment (Alxandr) [inline src/Authentication/Controllers/SystemRegisterController.cs:136]: Don't return a bool

File: src/Authentication/Controllers/SystemRegisterController.cs [hunk 3]
```diff
@@ -116,4 +116,40 @@ public async Task<ActionResult> CreateRegisteredSystem([FromBody] RegisterSystem
...
+    [HttpDelete("system/{systemId}")]
+    public async Task<ActionResult> SetDeleteOnRegisteredSystem(string systemId)
+    {
+        bool deleted = await _systemRegisterService.SetDeleteRegisteredSystemById(systemId);
+        if (!deleted) 
+        { 
+            return BadRequest(); 
+        }
+
+        return Ok(deleted);
```
Comment (Alxandr) [inline src/Authentication/Controllers/SystemRegisterController.cs:153]: Don't return bool

File: src/Authentication/Controllers/SystemRegisterController.cs [hunk 4]
```diff
@@ -116,4 +116,40 @@ public async Task<ActionResult> CreateRegisteredSystem([FromBody] RegisterSystem
...
+        return Ok(true);
+    }
+
+    /// <summary>
+    /// Set the registered system to be deleted.    
+    /// </summary>
+    /// <param name="systemId">The human readable string id</param>
+    /// <returns>true if changed</returns>
+    [HttpDelete("system/{systemId}")]
+    public async Task<ActionResult> SetDeleteOnRegisteredSystem(string systemId)
```
Comment (Alxandr) [inline src/Authentication/Controllers/SystemRegisterController.cs:145]: `ActionResult<T>`

**Review by Alxandr — APPROVED** [review]

File: src/Authentication/Controllers/SystemRegisterController.cs (hunk 1 above)
Comment (simen-rekkedal) [reply src/Authentication/Controllers/SystemRegisterController.cs:126]: ok, changed

File: src/Authentication/Controllers/SystemRegisterController.cs (hunk 2 above)
Comment (simen-rekkedal) [reply src/Authentication/Controllers/SystemRegisterController.cs:136]: ok, changed

File: src/Authentication/Controllers/SystemRegisterController.cs (hunk 4 above)
Comment (simen-rekkedal) [reply src/Authentication/Controllers/SystemRegisterController.cs:145]: ok, changed

File: src/Authentication/Controllers/SystemRegisterController.cs (hunk 3 above)
Comment (simen-rekkedal) [reply src/Authentication/Controllers/SystemRegisterController.cs:153]: ok, changed

**Pull request merged**